tests/validate_typecheck_testfile.testscript

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
```

Comment:
You could probably make this a commit hook instead. Or a sanity script that Travis runs.

Reply:
yes, but why? I think it is a desirable property that the test suite behaves exactly like what Travis does. This way I can run "make tests" before pushing and I'll know exactly what Travis will do and if it fails I can reproduce it 1:1 locally.

Reply:
I prefer the proposed approach rather than a hook -- this approach tests the state of a system as it is being tested via the CI, as opposed to the state of the system when a commit is made.

A sanity script that's run by Travis is pretty much what this is, but without tying it into a particular CI. While there's a bit of "hand-rolling" going on, I think that's fine here.